### ISSUE TITLE
Persistent 'unexpected token' error

### ISSUE THREAD

**Issue by jshier** [nicklockwood/SwiftFormat#1491]
Is there some sort of caching I need to clear to fix this? I have an error that's returned for the same token on slightly different lines where it doesn't exist.

> error: Unexpected token } at 114:5

In that file, there is no line 114 at all.

**Comment by nicklockwood** [comment]
You can try `swiftformat --cache clear`. I've never encountered any caching issues before though so it's a bit odd.

**Comment by jshier** [comment]
Clearing and ignoring doesn't seem to help. Currently I'm getting the error

> error: Unexpected token } at 101:5 in AppDomain.swift.

I've marked it in the code below. As you can see there's no `}` on that line, especially at character 5.

```swift
extension AppDomain {
    static func networkErrorAlert(_ error: any LocalizedError) -> AlertState<Action.Alert> {
        let title = "An Error Occurred"
        let message = """ // line 101
        Sorry, that network call failed due to an error, please try again.
        
        \(error.errorDescription ?? error.localizedDescription)
        """
        
        return .init(title: .init(title),
                     message: .init(message),
                     buttons: [.default(.init("OK"))])
    }
}
```

**Comment by nicklockwood** [comment]
@jshier were you ever able to resolve this? I'm still uncertain what might be causing it.

**Comment by jshier** [comment]
No, this is still occurring with 0.51.15. I can't see any unexpected invisibles in the file or anything that might trigger parsing oddness.

**Comment by jshier** [comment]
Okay, I've reproduced it more precisely and the issue isn't the code I posted. Here's the whole file with the content commented out that appears to cause the issue. Uncomment it and I see the error.

```
import ComposableArchitecture
import Foundation

struct AppDomain: Reducer {
    struct State: Equatable {
        enum UserState: Equatable {
            case signedIn(SignedInDomain.State)
            case signedOut(SignedOutDomain.State)
        }

        var user: UserState = .signedOut(.init())
        @PresentationState var alert: AlertState<Action.Alert>?
    }

    enum Action: Equatable {
        case openURL(URL)
        case signedIn(SignedInDomain.Action)
        case signedOut(SignedOutDomain.Action)
        case alert(PresentationAction<Alert>)

        enum Alert: Equatable {}
    }

    @Dependency(\.networking) var networking
    @Dependency(\.keychain) var keychain

    var body: some Reducer<State, Action> {
        // Doubly scoped to grab a particular enum value's case.
        Scope(state: \.user, action: /AppDomain.Action.signedOut) {
            Scope(state: /AppDomain.State.UserState.signedOut,
                  action: /.self) {
                SignedOutDomain()
            }
        }

        // Doubly scoped to grab a particular enum value's case.
        Scope(state: \.user, action: /AppDomain.Action.signedIn) {
            Scope(state: /AppDomain.State.UserState.signedIn,
                  action: /.self) {
                SignedInDomain()
            }
        }

        // App Reducer
//         Reduce { state, action in
//             switch action {
//             case .alert:
//                 return .none
//             case let .openURL(url):
//                 networking.handle(callbackURL: url)
//
//                 return .none
//             case let .signedOut(.receiveEmployee(employee)):
//                 state.user = .signedIn(.idle(for: employee))
//
//                 return .action(.signedIn(.refreshExpenses))
//             case .signedOut:
//                 // Handled by SignedOutDomain.
//                 return .none
//             case .signedIn(.logOut):
//                 networking.signOut()
//                 keychain.removeAll()
//                 state.user = .signedOut(.init())
//
//                 return .none
//             case .signedIn:
//                 // Handled by SignedInDomain.
//                 return .none
//             }
//         }
//         .ifLet(\.$alert, action: /Action.alert) // Automatically manage alert state.

        // Global Error Reducer
        Reduce { state, action in
            switch action {
            case let .signedOut(.receiveSignIn(.failure(error as any LocalizedError))),
                 let .signedOut(.receiveRawEmployee(_, .failure(error as any LocalizedError))),
                 let .signedIn(.receiveEmployee(.failure(error as any LocalizedError))),
                 let .signedIn(.receiveExpenses(.failure(error as any LocalizedError))):
                state.alert = AppDomain.networkErrorAlert(error)
            case .openURL, .signedIn, .signedOut, .alert:
                // Handled by other reducers.
                break
            }

            return .none
        }
    }
}

extension AppDomain {
    static func networkErrorAlert(_ error: any LocalizedError) -> AlertState<Action.Alert> {
        let title = "An Error Occurred"
        let message = """
        Sorry, that network call failed due to an error, please try again.

        \(error.errorDescription ?? error.localizedDescription)
        """

        return .init(title: .init(title),
                     message: .init(message),
                     buttons: [.default(.init("OK"))])
    }
}
```

**Comment by jshier** [comment]
It appears the issue is the comment on the `ifLet` line. The error is reproducible with just the `//` at the end of the line.

**Comment by nicklockwood** [comment]
@jshier I wasn't able to reproduce this on a file with those contents using swiftformat 0.51.15. Could you share your .swiftformat file? Maybe it's configuration-specific.

**Comment by jshier** [comment]
```
# file options

--symlinks ignore
--exclude Pods,Project/Generated
--swiftversion 5.8

# format options

--closingparen same-line
--commas inline
--comments indent
--decimalgrouping 3,5
--exponentcase lowercase
--exponentgrouping disabled
--fractiongrouping disabled
--ifdef no-indent
--importgrouping testable-top
--nospaceoperators ..<, ...
--someAny false
--stripunusedargs closure-only
--wraparguments preserve
--wrapcollections before-first

# rules

--enable isEmpty
--disable andOperator
--disable wrapMultilineStatementBraces
```

**Comment by nicklockwood** [comment]
@jshier I'm still not able to reproduce it with those exact file contents and configuration.

Are you able to reproduce this on another machine? Does it still happen if you copy and paste the contents of that file into another file? What if you copy the code you've pasted from github into a new file, etc?

My best guess at the moment is that it might be an invisible character in the file, or possibly something environmental on your machine.

**Comment by jshier** [comment]
Oh, did you uncomment the line that is marked as breaking? Just copy pasting the content direct from GitHub works fine, sorry about that.

**Comment by nicklockwood** [comment]
> Oh, did you uncomment the line that is marked as breaking? 

That was it, thanks 🤦‍♂️

**Comment by nicklockwood** [comment]
@jshier fixed in 0.52.1

**Comment by jshier** [comment]
Confirmed, thanks!
